ThreadLeakTest.testThreadLeakUtils failure

*maintenance* (commit bee5a9667eda77cd6a694337ca9b682413be831e)

Failed on Zulu JDK6: https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK6/lastCompletedBuild/testReport/classloading/ThreadLeakTest/testThreadLeakUtils/

Stacktrace:
```
java.lang.AssertionError: Timeout waiting for thread leaking-thread to terminate
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertJoinable(HazelcastTestSupport.java:933)
	at classloading.ThreadLeakTestUtils.assertHazelcastThreadShutdown(ThreadLeakTestUtils.java:68)
	at classloading.ThreadLeakTestUtils.assertHazelcastThreadShutdown(ThreadLeakTestUtils.java:58)
	at classloading.AbstractThreadLeakTest.assertThreadLeaks(AbstractThreadLeakTest.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:622)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:432)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at com.hazelcast.test.AbstractHazelcastClassRunner$3.evaluate(AbstractHazelcastClassRunner.java:335)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:379)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:340)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:413)
```

Standard output:
```
Started Running Test: testThreadLeakUtils
01:59:12,157 ERROR |testThreadLeakUtils| - [ThreadLeakTestUtils] testThreadLeakUtils - There should be one thread running!
-> leaking-thread (id: 1188602) (group: main) (daemon: true) (alive: true) (interrupted: false) (state: WAITING)
[sun.misc.Unsafe.park(Native Method), java.util.concurrent.locks.LockSupport.park(LockSupport.java:186), java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:838), java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998), java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304), java.util.concurrent.CountDownLatch.await(CountDownLatch.java:235), classloading.ThreadLeakTest$1.run(ThreadLeakTest.java:55)]

-> process reaper (id: 1188603) (group: main) (daemon: true) (alive: true) (interrupted: false) (state: RUNNABLE)
[]

THREAD DUMP FOR TEST FAILURE: "Expected exactly one running thread expected:<1> but was:<2>" at "testThreadLeakUtils"

"process reaper" 
	java.lang.Thread.State: RUNNABLE, cpu=-1 nsecs, usr=-1 nsecs, blocked=0 msecs, waited=0 msecs
		at java.lang.UNIXProcess.waitForProcessExit(Native Method)
		at java.lang.UNIXProcess.access$900(UNIXProcess.java:36)
		at java.lang.UNIXProcess$1$1.run(UNIXProcess.java:148)

"leaking-thread" 
	java.lang.Thread.State: WAITING, on lock=java.util.concurrent.CountDownLatch$Sync@7338c42d, cpu=0 nsecs, usr=0 nsecs, blocked=0 msecs, waited=559 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:838)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
		at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:235)
		at classloading.ThreadLeakTest$1.run(ThreadLeakTest.java:55)

"pool-86-thread-1" 
	java.lang.Thread.State: WAITING, on lock=java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@13d7a174, cpu=0 nsecs, usr=0 nsecs, blocked=0 msecs, waited=59815 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
		at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:386)
		at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1076)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1138)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
		at java.lang.Thread.run(Thread.java:704)

"hz._hzInstance_8356_dev.cached.thread-3" 
	java.lang.Thread.State: TIMED_WAITING, cpu=0 nsecs, usr=0 nsecs, blocked=0 msecs, waited=135428 msecs
		at java.lang.Thread.sleep(Native Method)
		at java.lang.Thread.sleep(Thread.java:321)
		at java.util.concurrent.TimeUnit.sleep(TimeUnit.java:356)
		at com.hazelcast.test.HazelcastTestSupport.sleepAtLeastMillis(HazelcastTestSupport.java:442)
		at com.hazelcast.test.HazelcastTestSupport.sleepAtLeastSeconds(HazelcastTestSupport.java:457)
		at com.hazelcast.executor.ExecutorServiceTestSupport$SleepingTask.call(ExecutorServiceTestSupport.java:163)
		at com.hazelcast.executor.ExecutorServiceTestSupport$SleepingTask.call(ExecutorServiceTestSupport.java:153)
		at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
		at java.util.concurrent.FutureTask.run(FutureTask.java:166)
		at com.hazelcast.executor.impl.DistributedExecutorService$CallableProcessor.run(DistributedExecutorService.java:285)
		at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1153)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
		at java.lang.Thread.run(Thread.java:704)
		at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
		at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)

"JitterThread" 
	java.lang.Thread.State: TIMED_WAITING, cpu=3560000000 nsecs, usr=1650000000 nsecs, blocked=0 msecs, waited=2228811 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:345)
		at com.hazelcast.test.jitter.JitterThread.sleepNanos(JitterThread.java:55)
		at com.hazelcast.test.jitter.JitterThread.run(JitterThread.java:39)

"RMI TCP Accept-21414" 
	java.lang.Thread.State: RUNNABLE, cpu=0 nsecs, usr=0 nsecs, blocked=0 msecs, waited=0 msecs
		at java.net.PlainSocketImpl.socketAccept(Native Method)
		at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:375)
		at java.net.ServerSocket.implAccept(ServerSocket.java:479)
		at java.net.ServerSocket.accept(ServerSocket.java:447)
		at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.executeAcceptLoop(TCPTransport.java:403)
		at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.run(TCPTransport.java:375)
		at java.lang.Thread.run(Thread.java:704)

"RMI TCP Accept-21413" 
	java.lang.Thread.State: RUNNABLE, cpu=0 nsecs, usr=0 nsecs, blocked=0 msecs, waited=0 msecs
		at java.net.PlainSocketImpl.socketAccept(Native Method)
		at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:375)
		at java.net.ServerSocket.implAccept(ServerSocket.java:479)
		at java.net.ServerSocket.accept(ServerSocket.java:447)
		at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.executeAcceptLoop(TCPTransport.java:403)
		at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.run(TCPTransport.java:375)
		at java.lang.Thread.run(Thread.java:704)

"surefire-forkedjvm-ping-30s" 
	java.lang.Thread.State: RUNNABLE, cpu=1290000000 nsecs, usr=960000000 nsecs, blocked=147 msecs, waited=2228583 msecs
		at java.io.FileInputStream.readBytes(Native Method)
		at java.io.FileInputStream.read(FileInputStream.java:239)
		at java.io.BufferedInputStream.read1(BufferedInputStream.java:273)
		at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
		at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:282)
		at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:324)
		at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:176)
		at java.io.InputStreamReader.read(InputStreamReader.java:184)
		at java.io.Reader.read(Reader.java:100)
		at java.util.Scanner.readInput(Scanner.java:797)
		at java.util.Scanner.findWithinHorizon(Scanner.java:1676)
		at java.util.Scanner.hasNextLine(Scanner.java:1493)
		at org.apache.maven.surefire.booter.PpidChecker$ProcessInfoConsumer.execute(PpidChecker.java:301)
		at org.apache.maven.surefire.booter.PpidChecker.unix(PpidChecker.java:160)
		at org.apache.maven.surefire.booter.PpidChecker.isProcessAlive(PpidChecker.java:115)
		at org.apache.maven.surefire.booter.ForkedBooter$2.run(ForkedBooter.java:213)
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
		at java.util.concurrent.FutureTask$Sync.innerRunAndReset(FutureTask.java:351)
		at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:178)
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:165)
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:267)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1153)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
		at java.lang.Thread.run(Thread.java:704)

"surefire-forkedjvm-command-thread" 
	java.lang.Thread.State: RUNNABLE, cpu=10000000 nsecs, usr=10000000 nsecs, blocked=0 msecs, waited=0 msecs
		at java.io.FileInputStream.readBytes(Native Method)
		at java.io.FileInputStream.read(FileInputStream.java:239)
		at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
		at java.io.BufferedInputStream.read(BufferedInputStream.java:254)
		at java.io.DataInputStream.readInt(DataInputStream.java:387)
		at org.apache.maven.surefire.booter.MasterProcessCommand.decode(MasterProcessCommand.java:115)
		at org.apache.maven.surefire.booter.CommandReader$CommandRunnable.run(CommandReader.java:391)
		at java.lang.Thread.run(Thread.java:704)

"Signal Dispatcher" 
	java.lang.Thread.State: RUNNABLE, cpu=0 nsecs, usr=0 nsecs, blocked=0 msecs, waited=0 msecs

"Finalizer" 
	java.lang.Thread.State: WAITING, on lock=java.lang.ref.ReferenceQueue$Lock@22c49037, cpu=560000000 nsecs, usr=530000000 nsecs, blocked=25 msecs, waited=2242061 msecs
		at java.lang.Object.wait(Native Method)
		at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:133)
		at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:149)
		at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:189)

"Reference Handler" 
	java.lang.Thread.State: WAITING, on lock=java.lang.ref.Reference$Lock@16d81cb6, cpu=100000000 nsecs, usr=80000000 nsecs, blocked=28 msecs, waited=2243035 msecs
		at java.lang.Object.wait(Native Method)
		at java.lang.Object.wait(Object.java:502)
		at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:133)


01:59:12,719 ERROR |testThreadLeakUtils| - [ThreadLeakTestUtils] main - There are still Hazelcast threads running after shutdown!
-> leaking-thread (id: 1188602) (group: main) (daemon: true) (alive: true) (interrupted: false) (state: WAITING)
[sun.misc.Unsafe.park(Native Method), java.util.concurrent.locks.LockSupport.park(LockSupport.java:186), java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:838), java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998), java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304), java.util.concurrent.CountDownLatch.await(CountDownLatch.java:235), classloading.ThreadLeakTest$1.run(ThreadLeakTest.java:55)]

02:04:12,721 ERROR |testThreadLeakUtils| - [ThreadLeakTestUtils] main - There are threads which survived assertJoinable()!
-> leaking-thread (id: 1188602) (group: main) (daemon: true) (alive: true) (interrupted: false) (state: WAITING)
[sun.misc.Unsafe.park(Native Method), java.util.concurrent.locks.LockSupport.park(LockSupport.java:186), java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:838), java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998), java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304), java.util.concurrent.CountDownLatch.await(CountDownLatch.java:235), classloading.ThreadLeakTest$1.run(ThreadLeakTest.java:55)]

BuildInfo right after testThreadLeakUtils(classloading.ThreadLeakTest): BuildInfo{version='3.10.5-SNAPSHOT', build='20180812', buildNumber=20180812, revision=bee5a96, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testThreadLeakUtils(classloading.ThreadLeakTest):'
01:59:10, accumulated pauses: 590 ms, max pause: 549 ms, pauses over 1000 ms: 0
01:59:15, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:20, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:25, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:30, accumulated pauses: 48 ms, max pause: 2 ms, pauses over 1000 ms: 0
01:59:35, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:40, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:45, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:50, accumulated pauses: 38 ms, max pause: 0 ms, pauses over 1000 ms: 0
01:59:55, accumulated pauses: 39 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:00:00, accumulated pauses: 37 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:00:05, accumulated pauses: 35 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:00:10, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:15, accumulated pauses: 40 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:20, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:25, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:30, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:35, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:40, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:45, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:00:50, accumulated pauses: 39 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:00:55, accumulated pauses: 33 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:00, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:05, accumulated pauses: 33 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:10, accumulated pauses: 34 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:15, accumulated pauses: 40 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:01:20, accumulated pauses: 38 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:01:25, accumulated pauses: 33 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:30, accumulated pauses: 38 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:01:35, accumulated pauses: 34 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:40, accumulated pauses: 30 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:01:45, accumulated pauses: 34 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:01:50, accumulated pauses: 34 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:01:55, accumulated pauses: 30 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:02:00, accumulated pauses: 30 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:02:05, accumulated pauses: 33 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:02:10, accumulated pauses: 31 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:02:15, accumulated pauses: 39 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:02:20, accumulated pauses: 37 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:02:25, accumulated pauses: 39 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:02:30, accumulated pauses: 33 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:02:35, accumulated pauses: 28 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:02:40, accumulated pauses: 34 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:02:45, accumulated pauses: 31 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:02:50, accumulated pauses: 33 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:02:55, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:00, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:05, accumulated pauses: 38 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:10, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:15, accumulated pauses: 38 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:20, accumulated pauses: 36 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:03:25, accumulated pauses: 40 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:30, accumulated pauses: 33 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:03:35, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:40, accumulated pauses: 38 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:45, accumulated pauses: 46 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:03:50, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:03:55, accumulated pauses: 40 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:04:00, accumulated pauses: 34 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:04:05, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:04:10, accumulated pauses: 19 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

--- a/hazelcast/src/test/java/classloading/ThreadLeakTestUtils.java
+++ b/hazelcast/src/test/java/classloading/ThreadLeakTestUtils.java
@@ -40,9 +40,13 @@ public final class ThreadLeakTestUtils {
      * List of whitelisted classes of threads, which are allowed to be not joinable.
      * We should not add classes of Hazelcast production code here, just test related classes.
      */
-    private static final List<Class> THREAD_CLASS_WHITELIST = asList(new Class[]{
+    private static final List<Class<?>> THREAD_CLASS_WHITELIST = asList(new Class<?>[]{
             JitterThread.class,
     });
+    private static final List<String> THREAD_NAME_WHITELIST = asList(
+            "process reaper",
+            "surefire-forkedjvm-ping-30s"
+    );
 
     private static final ILogger LOGGER = Logger.getLogger(ThreadLeakTestUtils.class);
 
@@ -100,7 +104,9 @@ public final class ThreadLeakTestUtils {
         Iterator<Thread> iterator = threads.iterator();
         while (iterator.hasNext()) {
             Thread thread = iterator.next();
-            if (THREAD_CLASS_WHITELIST.contains(thread.getClass())) {
+            Class<? extends Thread> threadClass = thread.getClass();
+            String threadName = thread.getName();
+            if (THREAD_CLASS_WHITELIST.contains(threadClass) || THREAD_NAME_WHITELIST.contains(threadName)) {
                 iterator.remove();
             }
         }

